### ISSUE TITLE
Linked block decompression (preset dictionaries) broken on Go 1.17

### ISSUE THREAD

**Issue by greatroar** [pierrec/lz4#136]
The amd64 assembler version of decodeBlock fails the linked block test with Go 1.17 on my Linux box:

```
$ go test
--- FAIL: TestReader (0.00s)
    --- FAIL: TestReader/testdata/Mark.Twain-Tom.Sawyer_linked.txt.lz4_[] (0.00s)
        reader_test.go:105: lz4: invalid source or destination buffer too short
        reader_test.go:109: invalid size: got 1000; want 4266361
        reader_test.go:113: uncompressed data does not match original
    --- FAIL: TestReader/testdata/Mark.Twain-Tom.Sawyer_linked.txt.lz4_[ConcurrencyOption(8)] (0.00s)
        reader_test.go:105: lz4: invalid source or destination buffer too short
        reader_test.go:109: invalid size: got 5000; want 4266361
        reader_test.go:113: uncompressed data does not match original
FAIL
exit status 1
FAIL	github.com/pierrec/lz4/v4	2.002s
```

The ARM and noasm versions are fine.

**Comment by greatroar** [comment]
It's broken with #133 too.

**Comment by greatroar** [comment]
Fixed on v4 by #136. @pierrec maybe make that the default branch for this repo?

**Comment by pierrec** [comment]
@greatroar thanks for all your contributions, really appreciated. I am thinking of making v4 the default branch. Will make it so soon(ish).
